scripts/tests

```diff
@@ -365,6 +368,58 @@ class State(object):
         return results
 
 
+"""
+Underlying functional tests
+
+"""
+
+
+def _func(state: State):
```

Comment:
It seems that `State` is a misnomer? An instance of `State` represents a particular build configuration, and offers commands to build and run tests under that configuration?

Could you use this PR to expand on the documentation please and rename `State` (or explain why this is the best choice of name)?

Reply:
The naming was a mistake like you mentioned... 😢 
I actually had it fixed in other commit, and planned to open a separate pr for this.
Are you ok to have this PR to focus on #251 first ?